### DISCUSSION TITLE
Issues with games created using 'Game Rules' & 'Hazard Map' dropdowns

### DISCUSSION THREAD

**Discussion by mawilson** [BattlesnakeOfficial/feedback#186] [Bug Reports]
I'm seeing some buggy behavior when creating games through the battlesnake /games/create page:
1. Game created with Game Rules 'Battlesnake Royale' & either Hazard Map ('Spiral' or 'Scatter') results in normal Royale hazards spawning along the edges, which do hazard damage as expected. However, in addition, every three turns an individual hazard will spawn on the map, & then immediately disappear the next turn. This hazard can even spawn in existing hazard, which results in a darker grey square on board, & two identical hazard coordinates in the gameState. Example game: https://play.battlesnake.com/g/101c478b-e708-48b4-9ef3-8b57fc41d83e/
2. Game created with Game Rules 'Wrapped' & either Hazard Map ('Spiral' or 'Scatter') results in the map generating hazards as expected, but the hazards doing no additional damage. The gameState shows gameState.game.settings.royale.shrinkEveryNTurns & gameState.game.settings.hazardDamagePerTurn both to be 0. Example game with spiral: https://play.battlesnake.com/g/ff3179a5-6015-4902-a7c4-2026a57c7878/. Example game with scatter: https://play.battlesnake.com/g/bde4c095-5c0d-4630-a277-0b3a9f2283bb/

Games created using Game Rules 'Battlesnake Royale' & Hazard Map 'No Hazards' seem to behave as expected.

**Comment by bvanvugt** [comment]
Thanks for reporting this! We just rolled out some fixes, here's what we did:

1. **We've disabled hazard maps when creating "Battlesnake Royale" games.** This was the original intent, but the disabling code was not working correctly. Battlesnake Royale and Hazard Maps do not work well together in the current implementation (and to some extent, don't make much sense when combined). In the future, Battlesnake Royale will shift to being defined as a Hazard Map rather than an independent game mode.
2. **We've fixed a bug that was preventing spiral and scatter hazard maps from applying the correct damage.** The scope of this bug was limited to custom games - it does not impact spring league, global arena, or other competitive play.

**Comment by mawilson** [comment]
Thanks for the quick response! I see this marked as 'shipped' so maybe I should create another discussion, but I am having another, possibly related issue with creating games now. When attempting to create a game with Game Rules 'Battlesnake Royale', most of the time I receive a red error banner up top reading 'Unable to create a game.' I have been able to successfully create a game at least once in this manner, but unfortunately haven't been able to reproduce that since - almost every time I get the red error banner, & can't create the game.
![UnableToCreateRoyale](https://user-images.githubusercontent.com/2830497/155903114-f853ff01-ca2a-40d9-8b42-a22c62ce8033.png)

**Reply by bvanvugt** [reply]
I was able to reproduce this, I'll see if I can get a fix out tomorrow :+1:

**Reply by bvanvugt** [reply]
@mawilson we just deployed a fix for this, if you want to give it another go. I'll hold on tagging this resolved until you've had success with it :+1:

**Reply by mawilson** [reply]
Looks good! No more issues creating normal Battlesnake Royale games, nor games with spiral or scatter Hazard Maps. I notice the 'Hazard Map' dropdown is now missing when using Battlesnake Royale rules, which also makes sense.
